ngraph_bridge/pass/transpose_sinking.cc

```diff
@@ -0,0 +1,504 @@
+                 << " input index " << input_index;
+  auto arg = target->input(input_index).get_source_output();
+  NGRAPH_VLOG(4) << "Arg shape: " << arg.get_shape();
+  auto new_order = ngraph::as_type_ptr<ngraph::opset3::Constant>(
+      transpose->input_value(1).get_node_shared_ptr());
+  auto new_transpose = make_transpose(arg.get_node_shared_ptr(),
+                                      new_order->get_axis_vector_val());
+  NGRAPH_VLOG(4) << "Inserting transpose "
+                 << describe<ngraph::opset3::Transpose>(new_transpose)
+                 << " at input " << target->get_name() << " input index "
+                 << input_index;
+
+  auto transpose_users = transpose->get_users().size();
+  NGRAPH_VLOG(4) << "Users " << transpose->get_users().size();
+  if (transpose_users > 1) {
+    if (reuse_map.find(transpose) != reuse_map.end()) {
+      // use the existing transpose
+      auto existing_transpose = reuse_map.at(transpose);
+      NGRAPH_VLOG(4) << "Using existing transpose "
+                     << describe<ngraph::opset3::Transpose>(existing_transpose);
+      target->input(input_index)
+          .replace_source_output(existing_transpose->output(0));
+    } else {
+      NGRAPH_VLOG(4) << "Write ReuseMap[" << transpose->get_name() << "] = "
+                     << describe<ngraph::opset3::Transpose>(new_transpose);
+      // update reuse_map
+      reuse_map[transpose] = new_transpose;
+      NGRAPH_VLOG(4) << "Using new transpose "
+                     << describe<ngraph::opset3::Transpose>(new_transpose);
+      target->input(input_index)
+          .replace_source_output(new_transpose->output(0));
+    }
+  } else {
+    target->input(input_index).replace_source_output(new_transpose->output(0));
+  }
+}
+
+static void delete_transpose(shared_ptr<ngraph::Node> transpose) {
+  NGRAPH_VLOG(4) << "Removing transpose " << transpose->get_name();
+  if (!transpose->get_users().empty()) {
+    ngraph::replace_node(transpose, transpose->get_argument(0));
+  }
+}
+
+static void mark_transpose_for_deletion(
+    shared_ptr<ngraph::Node> transpose,
+    set<shared_ptr<ngraph::Node>>& transposes_to_delete) {
+  NGRAPH_VLOG(4) << "Marking transpose " << transpose->get_name()
+                 << " for deletion";
+  transposes_to_delete.insert(transpose);
+}
+
+static shared_ptr<ngraph::opset3::Transpose> create_default_transpose(
+    shared_ptr<ngraph::Node> n) {
+  auto default_order = ngraph::get_default_order(n->get_shape());
+  auto default_transpose = make_transpose(n, default_order);
+  NGRAPH_VLOG(4) << "Default transpose: "
+                 << describe<ngraph::opset3::Transpose>(default_transpose);
+  return default_transpose;
+}
+
+// convert_binary_to_default_order is used when one of the arguments
+// of a binary op isn't in the default format (i.e. nhwc instead of nchw)
+// We have to normalize this other argument to nchw by swimming nchw towards
+// parameters
+// as far as we can
```

Comment:
Can you add what right and left mean, helps in understanding?

Reply:
I reworded the comment to indicate which of the arguments gets transposed/reshaped.